apps/docs/src/components/code/ExpandableLayout.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react";
+import { CodeEditor, useLiveRunner, type Scope } from "react-live-runner";
+import useEditorTheme from "@docs/hooks/useEditorTheme";
+
+import { ExpandableControls } from "./ExpandableControls";
+
+type ExpandableLayoutProps = {
+  scope: Scope | null;
+  code: Record<string, string>;
+};
+
+/**
+ * ExpandableLayout renders a live preview and an expandable code editor
+ * with interactive controls for toggling and resetting the code.
+ */
+export const ExpandableLayout = ({ scope, code }: ExpandableLayoutProps) => {
+  const editorTheme = useEditorTheme();
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const {
+    element,
+    code: editorCode,
+    error: liveError,
+    onChange,
+  } = useLiveRunner({
+    initialCode: Object.values(code)[0],
+    scope: scope || {},
+  });
+
+  const resetCode = () => onChange(Object.values(code)[0]);
+
+  return (
+    <div className="relative mt-3">
+      {/* Compact Controls */}
+      <ExpandableControls
+        onToggle={() => setIsExpanded((prev) => !prev)}
+        isExpanded={isExpanded}
+        code={editorCode}
+        onReset={resetCode}
+      />
+
+      {/* Preview Section */}
+      <div
+        className={[
+          "p-4 pt-5 [&_tr]:table-row gap-2 bg-[var(--uikit-colors-atmo1)] overflow-hidden",
```

Comment:
We keep should keep this as `atmo2` (`bgPage`). `atmo1` won't work for "surface" components (eg. `HvPanel`, `HvCard`, `HvSection`, etc.)

![image](https://github.com/user-attachments/assets/0247fe63-0325-4900-8e7a-9b99428b23b2)